gateway/src/apicast/policy/rate_limit/rate_limit.lua

```diff
@@ -151,7 +154,22 @@ end
 
 local function interpolate_key_name(context, limiters)
   for _, limiter in ipairs(limiters) do
-    limiter.key_name = limiter.template_string:render(context)
+    local m, _ = match(limiter.key.name, "{{ *(ngx\\.[^{}]+)}}")
+    if m then
+      local key_name = limiter.key.name
+      while m do
+        local replace = assert(loadstring(format("return %s", m[1])))()
+        local newstr, n, gsuberr = gsub(key_name, m[0], replace)
+        if gsuberr or n <= 0 then
+          break
+        end
+        key_name = newstr
+        m, _ = match(key_name, "{{ *(ngx\\.[^{}]+)}}")
+      end
+      limiter.key_name = key_name
+    else
+      limiter.key_name = limiter.template_string:render(context)
```

Comment:
If we don't use the regex, we have to have the list like `uri = ngx.var.uri` or `host = ngx.var.host` and so on.
The best place is `liquid_filters` of `template_string.lua`, isn't it?
Do you have any idea to avoid the `"API disabled in the current context"` error?

Reply:
I don't think `liquid_filters` in `template_string.lua` is the best place for this. Those are filters, functions like `md5`  that allow us to have things like  `{{ 'a_value' | md5 }}` in the policy config.

But `uri` and `host` are values, not filters. I think `uri` and `host` need to be included in the context passed to `render`. For that, I think a possible solution is to expose a table in `liquid_templates` with the values (not filters) that we want to expose. Something like:
```lua
_M.values = {
  uri = ngx.var.uri,
  host = ngx.var.host,
  -- possibly many others
}
```
The policies would be responsible for merging the policies context with this table and pass it to the context.

Reply:
Changed.